src/modules/connection.js

```diff
@@ -71,14 +72,29 @@ class Connection {
 
         // on error we remove the channel so the next call will recreate it (auto-reconnect are handled by connection users)
         channel.on('close', () => { delete connection.chann; });
-        channel.on('error', this._config.transport.error);
+        channel.on('error', this._onError.bind(this));
 
         connection.chann = channel;
         return channel;
       });
     return connection.chann;
   }
 
+  /**
+   * Log errors from connection/channel error events.
+   * @param {Error} error
+   */
+  _onError(error) {
+    this._config.transport.error(error);
+    if (this._config.logger) {
+      this._config.logger({
+        level: logLevels.Error,
```

Comment:
why we are passing `level` and not doing `this._config.logger.error` instead?

Reply:
I thought it would be simpler to handle a generic log event, since we will need to pass just one log function instead of passing duplicates of the same log function with only different levels.

I'll change it to a log function by level, but I still want it to receive a log event as a parameter.